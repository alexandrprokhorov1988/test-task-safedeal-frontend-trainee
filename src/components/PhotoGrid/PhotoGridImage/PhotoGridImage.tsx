import React from 'react';
import style from './PhotoGridImage.module.scss';

type PhotoGridImageProps = {
  /**
   * Image id.
   */
  id: number;
  /**
   * Image url.
   */
  url: string;
  /**
   * Image handleClick.
   */
  onImageClick: (id: number) => void;
}

/**
 * Image component.
 */
const PhotoGridImage: React.FC<PhotoGridImageProps> = ({id, url, onImageClick}) => {

  function handleImageClick(): void {
    onImageClick(id);
  }

  function handleKeyDown(e: React.KeyboardEvent): void | boolean {
    if (e.key === 'Enter') {
      onImageClick(id);
    }
    return false;
  }

  return (
    <img
      key={id}
      src={url}
      alt="Картинка"
      className={style.photoGridImage}
      onClick={handleImageClick}
      onKeyDown={handleKeyDown}
      tabIndex={0}
      role="presentation"
    />
  );
};

export default React.memo(PhotoGridImage);
