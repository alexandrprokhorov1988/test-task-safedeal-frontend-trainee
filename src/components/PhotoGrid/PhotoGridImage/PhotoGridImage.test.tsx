import React from 'react';
import {render, unmountComponentAtNode} from "react-dom";
import renderer from 'react-test-renderer';
import {act} from "react-dom/test-utils";
import PhotoGridImage from './PhotoGridImage';

let container: any = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Image component', () => {
  it('Должна вызывать коллбэк функцию по click', () => {
    const onChange = jest.fn();
    act(() => {
      render(<PhotoGridImage id={237} url="#" onOpen={onChange}/>, container);
    });
    const image = document.querySelector("img");
    act(() => {
      image?.dispatchEvent(new MouseEvent("click", {bubbles: true}));
    });
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('Должна вызывать коллбэк функцию по enter', () => {
    const onChange = jest.fn();
    act(() => {
      render(<PhotoGridImage id={237} url="#" onOpen={onChange}/>, container);
    });
    const image = document.querySelector("img");
    act(() => {
      image?.dispatchEvent(new KeyboardEvent("keydown", {bubbles: true, key: 'Enter'}));
    });
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('Image snapshot', () => {
    const result = renderer
      .create(<PhotoGridImage id={237} url="#" onOpen={() => {
      }}/>)
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
