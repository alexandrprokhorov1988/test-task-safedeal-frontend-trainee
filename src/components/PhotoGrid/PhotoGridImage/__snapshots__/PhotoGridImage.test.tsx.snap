// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Image snapshot 1`] = `
<img
  alt="Картинка"
  className="photoGridImage"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
  src="#"
  tabIndex={0}
/>
`;
